client/my-sites/plans-features-main/index.tsx

```diff
@@ -754,6 +757,52 @@ const PlansFeaturesMain = ( {
 		</div>
 	);
 
+	const usePlanSlugsForAllDisplayedIntervals = () => {
+		const currentPlanSlugs = gridPlansForFeaturesGrid?.map( ( { planSlug } ) => planSlug );
+
+		return currentPlanSlugs?.flatMap( ( planSlug ) =>
+			filteredDisplayedIntervals
+				.map( ( term ) =>
+					getPlanSlugForTermVariant( planSlug, URL_FRIENDLY_TERMS_MAPPING[ term ] )
+				)
+				.filter( ( planSlug ) => planSlug !== undefined )
+		);
+	};
+
+	const pricingForAllDisplayedIntervals = Plans.usePricingMetaForGridPlans( {
+		planSlugs: usePlanSlugsForAllDisplayedIntervals() ?? [],
+		storageAddOns,
+		coupon,
+		siteId,
+		useCheckPlanAvailabilityForPurchase,
+	} );
+
+	const enableTermSavingsPriceDisplay = useMemo( () => {
+		const isAnyGridPlanDiscounted = Object.values( pricingForAllDisplayedIntervals ?? {} ).reduce(
+			( isDiscounted, { discountedPrice, introOffer } ) => {
+				const hasDiscount =
+					'number' === typeof discountedPrice.monthly ||
+					( introOffer && ! introOffer.isOfferComplete );
+				return isDiscounted || !! hasDiscount;
+			},
+			false
+		);
+
+		if ( isAnyGridPlanDiscounted ) {
+			return false;
+		}
+
+		return (
+			( isEnabled( 'plans/term-savings-price-display' ) ||
+				longerPlanTermDefaultExperiment.isEligibleForTermSavings ) &&
+			isInSignup
+		);
+	}, [
+		pricingForAllDisplayedIntervals,
+		longerPlanTermDefaultExperiment.isEligibleForTermSavings,
+		isInSignup,
+	] );
+
```

Comment:
@jeyip this is one of the alternatives I had in mind. Does it look more sane from previously?

Complexity is still of the same order (processing all plans), but a smaller surface area of the changes (not touching hooks, etc.). Next step would be to clean this up into a separate hook so it doesn't live like this in the component.

Reply:
> this is one of the alternatives I had in mind. Does it look more sane from previously?

From where we were going before? The smaller surface area is absolutely better 👍 

Currently reading the code to gauge my understanding.

Reply:
> Currently reading the code to gauge my understanding.

Yes this is significantly more easy to follow 🎉 

> Next step would be to clean this up into a separate hook so it doesn't live like this in the component.

Totally agree on encapsulating this complexity.

I suspect you might've already thought about this, but it also seems like an eventual refactor ( doesn't have to be done soon ) would be for the new hook to handle `isAnyPlanPriceDiscounted` logic in the `Header-Price` component

https://github.com/Automattic/wp-calypso/blob/63d3c4a4da3e8320e71962c64534562006cc2ab3/packages/plans-grid-next/src/components/shared/header-price/index.tsx#L73-L97

Reply:
Ok - so I moved it now to a separate hook.

> I suspect you might've already thought about this, but it also seems like an eventual refactor ( doesn't have to be done immediately imo ) would be for the new hook to handle isAnyPlanPriceDiscounted logic in the Header-Price component.

Indeed, I had the same thinking. However, we can maybe excuse the two co-existing in that they serve different purposes - one being a header UI concern (that the header price has indeed rendered a discounted price), the other being a general flag into the system/app to enable/disable other things. I think a good refactor might be to turn it into a hook in the plans data-store. I will explore that in a follow-up.

Reply:
> Ok - so I moved it now to a separate hook.

Wonderful! Now we can also explore any necessary performance optimizations in one place ( if necessary ).

Reply:
Awesome. I'll create an issue for the above in a bit.

Reply:
^ https://github.com/Automattic/wp-calypso/issues/97132